File: src/app/library/models/book.ts
import { Genre } from 'src/app/admin/models/genre';
import { Author } from 'src/app/admin/models/author';
import { BookStore } from 'src/app/admin/models/book-store';

export class Book {
    bookId: number;
    title: string;
    isbn: string;
    imageUrl: string;
    year: Date;
    editura: string;
    description: string;
    nrOfPages: number;
    buyPrice: number;
    language: string;
    rate: number;
    wish: boolean;
    loan: string;
    genres: Genre[];
    authors: Author[];
    stores: BookStore[];
}

File: Models/Library/Book.cs
using BibliotecaOnline.DbModels;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace BibliotecaOnline.Models.Library
{
    public class Book
    {
        public int BookId { get; set; }
        public string Title { get; set; }
        public string ImageUrl { get; set; }
        public double Rate { get; set; }

        public string Loan { get; set; }
        public bool Wish { get; set; }

        public List<Genre> Genres { get; set; }
        public List<Author> Authors { get; set; }
        public List<BookStore> Stores { get; set; }
    }
}
